webpack.config.js

```diff
@@ -315,33 +75,295 @@ function devServerProxyBypass({ path }) {
+            }, {
+                test: /\/react\/features\/base\/util\/react-focus-lock-wrapper.js$/,
+                resolve: {
+                    alias: {
+                        'react-focus-lock': `${__dirname}/node_modules/react-focus-lock`
+                    }
+                }
+            }, {
+                test: /\.svg$/,
+                use: [ {
+                    loader: '@svgr/webpack',
+                    options: {
+                        dimensions: false,
+                        expandProps: 'start'
+                    }
+                } ]
+            } ]
+        },
+        node: {
+            // Allow the use of the real filename of the module being executed. By
+            // default Webpack does not leak path-related information and provides a
+            // value that is a mock (/index.js).
+            __filename: true
+        },
+        optimization: {
+            concatenateModules: minimize,
+            minimize
+        },
+        output: {
+            filename: `[name]${minimize ? '.min' : ''}.js`,
+            path: `${__dirname}/build`,
+            publicPath: '/libs/',
+            sourceMapFilename: `[name].${minimize ? 'min' : 'js'}.map`
+        },
+        plugins: [
+            detectCircularDeps
+                && new CircularDependencyPlugin({
+                    allowAsyncCycles: false,
+                    exclude: /node_modules/,
+                    failOnError: false
+                })
+        ].filter(Boolean),
+        resolve: {
+            alias: {
+                'focus-visible': 'focus-visible/dist/focus-visible.min.js'
+            },
+            aliasFields: [
+                'browser'
+            ],
+            extensions: [
+                '.web.js',
 
-    if (path.startsWith('/libs/')) {
-        return path;
-    }
+                // Webpack defaults:
+                '.js',
+                '.json'
+            ],
+            fallback: {
+                // Provide some empty Node modules (required by AtlasKit, olm).
+                crypto: false,
+                fs: false,
+                path: false,
+                process: false
+            }
+        }
+    };
+}
+
+/**
+ * Helper function to build the dev server config. It's necessary to split it in
+ * Webpack 5 because only one devServer entry is supported, so we attach it to
+ * the main bundle.
+ *
+ * @returns {Object} the dev server configuration.
+ */
+function getDevServerConfig() {
+    return {
+        https: true,
+        host: '127.0.0.1',
+        inline: true,
+        proxy: {
+            '/': {
+                bypass: devServerProxyBypass,
+                secure: false,
+                target: devServerProxyTarget,
+                headers: {
+                    'Host': new URL(devServerProxyTarget).host
+                }
+            }
+        }
+    };
 }
+
+module.exports = (_env, argv) => {
+    const analyzeBundle = Boolean(process.env.ANALYZE_BUNDLE);
```

Comment:
is there a reason we don't extract this up top rather than passing it to `getBundleAnalyzerPlugin` and `getPerformanceHints`?

Reply:
Both of those groups of options rely on knowing the mode (prod or dev) and in Webpack 5 that is accomplished by exporting a function instead and checking the argv object. Hence I extracted and passed down the options where necessary.

Reply:
```bash
263         },
264         plugins: [
265             ...config.plugins,
266             ...getBundleAnalyzerPlugin('analytics-ga')
267         ],
268         performance: getPerformanceHints(5 * 1024)
269     }),
```

cant find in `webpack.config.js`